### PR TITLE
refactor: add context parameter to all executor methods

### LINKED ISSUE #25: Add contexts to API calls
Original design omitted contexts in all methods, but they have to be added to the interface.

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

This commit adds context.Context as the first parameter to all executor methods and updates relevant function calls and mocks accordingly. This change enhances context propagation and allows for better control over request lifecycles and timeouts.

Resolves #25.

## Summary by CodeRabbit

- **New Features**
	- Enhanced method signatures across various components to support context-aware execution, improving control over cancellations and deadlines.

- **Bug Fixes**
	- Adjusted test cases to align with new method signatures, ensuring proper context management during execution.

- **Documentation**
	- Updated comments and method signatures to reflect the addition of context parameters, promoting better understanding and usage of the API.

### CODE REVIEW

**Review by gupadhyaya — DISMISSED** [review]

**Review by Manav-Aggarwal — DISMISSED** [review]

**Review by yarikbratashchuk — DISMISSED** [review]

**Review by Manav-Aggarwal — DISMISSED** [review]

File: proxy/grpc/server.go [hunk 1]
```diff
@@ -65,7 +61,7 @@
 
 // GetTxs handles GetTxs method call from execution API.
 func (s *Server) GetTxs(ctx context.Context, req *pb.GetTxsRequest) (*pb.GetTxsResponse, error) {
-	txs, err := s.exec.GetTxs()
+	txs, err := s.exec.GetTxs(context.TODO())
```
[the remark this answers is not in the document]
Comment (gupadhyaya) [reply proxy/grpc/server.go:64]: @tzdybal i think you should use the ctx here

File: proxy/grpc/server.go [hunk 2]
```diff
@@ -108,7 +105,7 @@ func (s *Server) ExecuteTxs(ctx context.Context, req *pb.ExecuteTxsRequest) (*pb
 
 // SetFinal handles SetFinal method call from execution API.
 func (s *Server) SetFinal(ctx context.Context, req *pb.SetFinalRequest) (*pb.SetFinalResponse, error) {
-	err := s.exec.SetFinal(req.BlockHeight)
+	err := s.exec.SetFinal(context.TODO(), req.BlockHeight)
```
[the remark this answers is not in the document]
Comment (gupadhyaya) [reply proxy/grpc/server.go:108]: same

File: proxy/grpc/server.go [hunk 3]
```diff
@@ -48,11 +48,7 @@
 	// Convert Unix timestamp to UTC time
 	genesisTime := time.Unix(req.GenesisTime, 0).UTC()
 
-	stateRoot, maxBytes, err := s.exec.InitChain(
-		genesisTime,
-		req.InitialHeight,
-		req.ChainId,
-	)
+	stateRoot, maxBytes, err := s.exec.InitChain(context.TODO(), genesisTime, req.InitialHeight, req.ChainId)
```
[the remark this answers is not in the document]
Comment (gupadhyaya) [reply proxy/grpc/server.go:51]: could use ctx instead of context.TODO() as suggested

**Review by MSevey — DISMISSED** [review]

**Review by yarikbratashchuk — DISMISSED** [review]

File: proxy/grpc/client_server_test.go [hunk 4]
```diff
@@ -45,7 +48,7 @@ func TestClientServer(t *testing.T) {
 	require.NoError(t, err)
 	defer func() { _ = client.Stop() }()
 
-	mockExec.On("GetTxs").Return([]types.Tx{}, nil).Maybe()
+	mockExec.On("GetTxs", mock.Anything).Return([]types.Tx{}, nil).Maybe()
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply proxy/grpc/client_server_test.go:51]: Created follow-up issue: https://github.com/rollkit/go-execution/issues/33

File: proxy/grpc/proxy_test.go [hunk 5]
```diff
@@ -65,7 +65,7 @@ func (s *ProxyTestSuite) SetupTest() {
 	require.NoError(s.T(), err)
 
 	for i := 0; i < 10; i++ {
-		if _, err := client.GetTxs(); err == nil {
+		if _, err := client.GetTxs(context.TODO()); err == nil {
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply proxy/grpc/proxy_test.go:68]: See https://github.com/rollkit/go-execution/issues/33

File: proxy/jsonrpc/server.go [hunk 6]
```diff
@@ -90,11 +91,7 @@ func (s *Server) handleInitChain(params json.RawMessage) (interface{}, *jsonRPCE
 		return nil, ErrInvalidParams
 	}
 
-	stateRoot, maxBytes, err := s.exec.InitChain(
-		time.Unix(p.GenesisTime, 0).UTC(),
-		p.InitialHeight,
-		p.ChainID,
-	)
+	stateRoot, maxBytes, err := s.exec.InitChain(context.TODO(), time.Unix(p.GenesisTime, 0).UTC(), p.InitialHeight, p.ChainID)
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply proxy/jsonrpc/server.go:94]: See: https://github.com/rollkit/go-execution/issues/34

**Review by MSevey — APPROVED** [review]

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**